src/mam4xx/aero_model.hpp

```diff
@@ -0,0 +1,626 @@
+    for (int ja = 0; ja < na; ++ja) {
+      // aerosol particle radius
+      // rain droplet and aerosol particle radius [cm]
+      const Real r_aer = raerosv[ja];
+      Real etotal = zero; // efficiency of total scavenging effects [fraction]
+      calc_impact_efficiency(r_aer, r_rain, temp,       // & ! in
+                             freepath, rhoaero, rhoair, // & ! in
+                             vfall, airkinvisc,         // & ! in
+                             etotal);                   // out
+
+      // rain droplet sweep out volume [cm3/cm3/s]
+      const Real rainsweepout =
+          xnumrainsv[jr] * 4 * pi * r_rain * r_rain * vfall;
+      scavsumnumbb += rainsweepout * etotal * fnumaerosv[ja];
+      scavsumvolbb += rainsweepout * etotal * fvolaerosv[ja];
+    } // ja_loop
+
+    scavsumnum += scavsumnumbb;
+    scavsumvol += scavsumvolbb;
+
+  } // jr_loop
+
+  scavratenum = scavsumnum * 3600;
+  scavratevol = scavsumvol * 3600;
+
+} // end calc_1_impact_rate
+
+KOKKOS_INLINE_FUNCTION
+void modal_aero_bcscavcoef_init(
+    const Real dgnum_amode[AeroConfig::num_modes()],
+    const Real sigmag_amode[AeroConfig::num_modes()],
+    const Real aerosol_dry_density[AeroConfig::num_modes()],
+    Real scavimptblnum[nimptblgrow_total][AeroConfig::num_modes()],
+    Real scavimptblvol[nimptblgrow_total][AeroConfig::num_modes()]) {
+  // -----------------------------------------------------------------------
+  //
+  //  Purpose:
+  //  Computes lookup table for aerosol impaction/interception scavenging rates
+  //
+  //  Authors: R. Easter
+  //
+  // -----------------------------------------------------------------------
+  // @param [in]   dgnum_amode aerosol diameters [m]
+  // @param [in]   sigmag_amode standard deviation of aerosol size distribution
+  // @param [out]  scavimptblnum scavenging rate of aerosol number [1/s]
+  // @param [out]  scavimptblnum scavenging rate of aerosol volume [1/s]
+  // @aerosol_dry_density [in] aerosol dry density [k/m3]
+  // FIXME : create an 4 elements array that contains the aerosol densities to
+  // replace specdens_amode and lspectype_amode
+
+  const Real zero = 0;
+  const Real one = 1;
+  // ! set up temperature-pressure pair to compute impaction scavenging rates
+  // BAD CONSTANT
+  const Real temp_0C = haero::Constants::melting_pt_h2o; //     ! K
+  const Real press_750hPa = 0.75e6;                      //  ! dynes/cm2
+  for (int imode = 0; imode < AeroConfig::num_modes(); ++imode) {
+    const Real sigmag = sigmag_amode[imode];
+    // Note: we replaced lspectype_amode and lspectype_amode for
+    // dry_aero_density const int ll = lspectype_amode[0][imode]; const Real
+    // rhodryaero = specdens_amode[ll];
+    const Real rhodryaero = aerosol_dry_density[imode];
+    for (int jgrow = nimptblgrow_mind; jgrow <= nimptblgrow_maxd; ++jgrow) {
+      // ratio of diameter for wet/dry aerosols [fraction]
+      const Real wetdiaratio = haero::exp(jgrow * dlndg_nimptblgrow);
```

Comment:
It could matter if we have a division with an integer. In this case dlndg_nimptblgrow is real. However, I convert `jgrow` to real inside the `exp` function.